src/components/stepforms/schemas/rollup.ts

```diff
@@ -0,0 +1,95 @@
+import { addNotInColumnNamesConstraint, StepFormType } from './utils';
+
+const schema = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  title: 'Rollup step',
+  type: 'object',
+  properties: {
+    name: {
+      type: 'string',
+      enum: ['rollup'],
+    },
+    hierarchy: {
+      type: 'array',
+      items: {
+        type: 'string',
+        minLength: 1,
+      },
+      title: 'hierarchal columns',
+      description: 'The hierarchy of columns, from lowest to highest level',
+      attrs: {
+        placeholder: 'Add columns',
+      },
+    },
+    aggregations: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          column: {
+            type: 'string',
+            minLength: 1,
+          },
+          aggfunction: {
+            type: 'string',
+            enum: ['sum', 'avg', 'count', 'min', 'max'],
+          },
+          newcolumn: {
+            type: 'string',
+            minLength: 1,
+          },
+        },
+        required: ['column', 'aggfunction', 'newcolumn'],
+        additionalProperties: false,
+      },
+      minItems: 1,
+      title: 'Aggregations',
+      description: 'The aggregations to be performed',
+    },
+    groupby: {
+      type: 'array',
+      items: {
+        type: 'string',
+        minLength: 1,
+      },
+      title: 'Group by',
+      description: 'The columns used to group rollup computation',
+      attrs: {
+        placeholder: 'Add columns',
+      },
+    },
+    labelCol: {
+      type: 'string',
+      minLength: 1,
+      title: 'Label column name',
+      description: 'The name of the label column that will be created',
+      attrs: {
+        placeholder: 'label',
+      },
+    },
+    levelCol: {
+      type: 'string',
+      minLength: 1,
+      title: 'Level column name',
+      description: 'The name of the level column that will be created',
+      attrs: {
+        placeholder: 'level',
+      },
+    },
+    parentCol: {
+      type: 'string',
+      minLength: 1,
+      title: 'Parent column name',
+      description: 'The name of the parent column that will be created',
+      attrs: {
+        placeholder: 'parent',
+      },
+    },
+  },
+  required: ['name', 'hierarchy', 'aggregations'],
+  additionalProperties: false,
+};
+
+export default function buildSchema(form: StepFormType) {
+  return addNotInColumnNamesConstraint(schema, 'aggregations.items.newcolumn', form.columnNames);
```

Comment:
:exclamation: 
Shouldn't we also do this operation with `labelCol`, `levelCol` and `parentCol`?

Reply:
Actually, I will delete this part even for aggregations. In this step we recreate the whole data structure, so there is no risk to delete an existing column... The only risk is to give the same name for the label column and the level column for exemple, but well... If you think it is important to prevent this, then we should overload the validate method of the base class StepForm.vue, as the `addNotInColumnNamesConstraint` checks unicity only against existing columns, not against other columns to be created.